contrib/node/tests/python/pants_test/contrib/node/tasks/test_apache_thrift_node_gen.py

```diff
@@ -0,0 +1,131 @@
+                            target_type=NodeThriftLibrary,
+                            sources=['test.thrift'])
+    synthetic_target = self.generate_single_thrift_target(test)
+    self.assertEqual({'package.json',
+                      'yarn.lock',
+                      'test_types.js'},
+                     set(synthetic_target.sources_relative_to_target_base()))
+
+  def test_nested_namespace(self):
+    self.create_file('src/thrift/com/foo/test1.thrift', contents=dedent("""
+    namespace js gen.foo.bar
+
+    struct Test1 {}
+    """))
+    self.create_file('src/thrift/com/foo/test2.thrift', contents=dedent("""
+    namespace js gen.foo.bar
+
+    struct Test2 {}
+    """))
+    test1 = self.make_target(spec='src/thrift/com/foo:test1',
+                             target_type=NodeThriftLibrary,
+                             sources=['test1.thrift', 'test2.thrift'])
+    synthetic_target = self.generate_single_thrift_target(test1)
+    self.assertEqual({'package.json',
+                      'yarn.lock',
+                      'test1_types.js',
+                      'test2_types.js'},
+                     set(synthetic_target.sources_relative_to_target_base()))
+
+  def test_namespace_effective(self):
+    self.create_file('src/thrift/com/foo/test1.thrift', contents=dedent("""
+    namespace js gen.foo.bar
+
+    struct Test1 {}
+    """))
+    self.create_file('src/thrift/com/foo/test2.thrift', contents=dedent("""
+    namespace js gen.foo.bar
+
+    struct Test2 {}
+    """))
+    test1 = self.make_target(spec='src/thrift/com/foo:test1',
+                             target_type=NodeThriftLibrary,
+                             sources=['test1.thrift'])
+    test2 = self.make_target(spec='src/thrift/com/foo:test2',
+                             target_type=NodeThriftLibrary,
+                             sources=['test2.thrift'])
+    synthetic_target1 = self.generate_single_thrift_target(test1)
+    synthetic_target2 = self.generate_single_thrift_target(test2)
+
+    self.assertNotEqual(synthetic_target1, synthetic_target2)
+
+  def test_thrift_target_dependable(self):
+    self.create_file('src/thrift/com/foo/test1.thrift', contents=dedent("""
+    namespace js gen.foo.bar
+
+    struct Test1 {}
+    """))
+    test1 = self.make_target(spec='src/thrift/com/foo:test1',
+                             target_type=NodeThriftLibrary,
+                             sources=['test1.thrift'])
+
+    self.create_file('src/thrift/com/foo/test2.thrift', contents=dedent("""
+    namespace js gen.foo.bar
+    include "test1.thrift"
+
+    struct Test2 {
+      1: required test1.Test1 test_obj;
+    }
+    """))
+
+    test2 = self.make_target(spec='src/thrift/com/foo:test2',
+                             target_type=NodeThriftLibrary,
+                             sources=['test2.thrift'],
+                             dependencies=[test1])
+    synthetic_node_target1, synthetic_node_target2 = self.generate_multiple_thrift_targets(test2)
```

Comment:
```
synthetic_node_targets = self.generate_multiple_thrift_targets(test2)
# check if len(2)
# then run tests
...
self.assertEqual({'package.json',
                      'yarn.lock',
                      'test1_types.js'},
                     set(synthetic_node_target[0].sources_relative_to_target_base()))
```

Reply:
Also are the orders guaranteed for the targets returned?

Reply:
yes since we're returning a list and not an unordered set so order is same.